src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Setup/ConfiguredSetupComponentSnapshot.cs

```diff
@@ -1,72 +1,56 @@
     {
-        _suggestedWorkloadComponentIds = ComponentIds = ImmutableStringHashSet.EmptyVisualStudioSetupComponentIds;
-        IsEmpty = true;
+        _requiresWebComponent = requiresWebComponent;
+        ComponentIds = componentIds;
     }
-    
-    private ConfiguredSetupComponentSnapshot(bool requiresWebComponent, string? netCoreTargetFrameworkVersion, ImmutableHashSet<string> suggestedWorkloadComponentIds)
+
+    /// <summary>
+    /// Applies changes to the current (immutable) snapshot, producing a new snapshot.
+    /// </summary>
+    /// <returns>The updated snapshot, or the same instance if no changes were required.</returns>
+    public ConfiguredSetupComponentSnapshot Update(IProjectSubscriptionUpdate buildUpdate, IProjectCapabilitiesSnapshot capabilities)
     {
-        _requiresWebComponent = requiresWebComponent;
-        _netCoreTargetFrameworkVersion = netCoreTargetFrameworkVersion;
-        _suggestedWorkloadComponentIds = suggestedWorkloadComponentIds;
+        ImmutableHashSet<string> componentIds;
 
-        ImmutableHashSet<string> componentIds = suggestedWorkloadComponentIds;
+        bool requiresWebComponent;
 
-        if (requiresWebComponent)
-        {
-            componentIds = componentIds.Add(SetupComponentReferenceData.WebComponentId);
-        }
+        ProcessCapabilities();
+        ProcessBuildUpdate();
 
-        if (netCoreTargetFrameworkVersion is not null && SetupComponentReferenceData.TryGetComponentIdByNetCoreTargetFrameworkVersion(netCoreTargetFrameworkVersion, out string? runtimeComponentId))
+        if (ReferenceEquals(componentIds, ComponentIds))
         {
-            componentIds = componentIds.Add(runtimeComponentId);
+            return this;
         }
 
-        ComponentIds = componentIds;
-    }
+        return new(requiresWebComponent, componentIds);
 
-    public ConfiguredSetupComponentSnapshot Update(
-        IProjectSubscriptionUpdate evaluationUpdate,
-        IProjectSubscriptionUpdate buildUpdate,
-        IProjectCapabilitiesSnapshot capabilities)
-    {
-        // We use bitwise | here instead of logical || to prevent short circuiting.
-        if (IsEmpty |
-            ProcessCapabilities(out bool requiresWebComponent) |
-            ProcessEvaluationUpdate(out string? netCoreTargetFrameworkVersion) |
-            ProcessBuildUpdate(out ImmutableHashSet<string> suggestedWorkloadComponentIds))
+        void ProcessCapabilities()
         {
-            return new(requiresWebComponent, netCoreTargetFrameworkVersion, suggestedWorkloadComponentIds);
-        }
-
-        return this;
+            const string webComponentId = "Microsoft.VisualStudio.Component.Web";
 
-        bool ProcessCapabilities(out bool requiresWebComponent)
-        {
             requiresWebComponent = RequiresWebComponent();
-            return requiresWebComponent != _requiresWebComponent;
+
+            componentIds = (requiresWebComponent, _requiresWebComponent) switch
```

Comment:
What do you think about disambiguating these variable names more?

Reply:
Can you give an example, either of better names, or which names you find ambiguous?